packages/block-library/src/missing/test/edit-integration.native.js

```
@@ -1,81 +1,171 @@
 /**
  * External dependencies
  */
-import { initializeEditor, fireEvent, within } from 'test/helpers';
+import {
+	fireEvent,
+	getBlock,
+	initializeEditor,
+	screen,
+	setupCoreBlocks,
+	withFakeTimers,
+	within,
+} from 'test/helpers';
+import { Platform } from 'react-native';
 
 /**
  * WordPress dependencies
  */
-import { getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
+import { unregisterBlockType } from '@wordpress/blocks';
 import { setLocaleData } from '@wordpress/i18n';
+import { requestUnsupportedBlockFallback } from '@wordpress/react-native-bridge';
+
+// Override modal mock to prevent unmounting it when is not visible.
+// This is required to be able to trigger onClose and onDismiss events when
+// the modal is dismissed.
+jest.mock( 'react-native-modal', () => {
+	const mockComponent = require( 'react-native/jest/mockComponent' );
+	return mockComponent( 'react-native-modal' );
+} );
```

IIRC we have done something similar elsewhere. At some point, we may just define a more robust default mock to avoid the need for these one-offs. We do not necessarily need to do that in this PR.

---

Yeah, we needed to do the same in the integration tests of the Embed block ([reference](https://github.com/WordPress/gutenberg/blob/404518fa4c0aa8dc3eedf40afd4c7602461dce6a/packages/block-library/src/embed/test/index.native.js#L35-L41)). I agree we could try to unify the behavior of the Modal mock, currently the default one is to unmount the component when hides:
https://github.com/WordPress/gutenberg/blob/404518fa4c0aa8dc3eedf40afd4c7602461dce6a/test/native/setup.js#L139-L143

I presume this behavior is the easiest approach to assert in a test if a modal is hidden. Now that we have the [`toBeVisible` matcher](https://github.com/WordPress/gutenberg/blob/404518fa4c0aa8dc3eedf40afd4c7602461dce6a/test/native/matchers/to-be-visible.js), we could consider dropping this default behavior and simply mock the modal component. I can work on this in a separate PR.